libbeat/processors/add_kubernetes_metadata/kubernetes.go

```diff
@@ -316,19 +313,16 @@ func (k *kubernetesAnnotator) Run(event *beat.Event) (*beat.Event, error) {
 		return event, nil
 	}
 	if kubernetesMetadataExist(event) {
-		k.log.Debug("Skipping add_kubernetes_metadata processor as kubernetes metadata already exist")
 		return event, nil
 	}
+
 	index := k.matchers.MetadataIndex(event.Fields)
 	if index == "" {
-		k.log.Debug("No container match string, not adding kubernetes data")
```

Comment:
Maybe this can stay. I quess is a valid scenario to understand why sth did not have metadata and troubleshoot

Reply:
From my understanding [MetadataIndex](https://github.com/elastic/beats/blob/990bd5b1279cdab4552f5c35d6adf52131550bd2/libbeat/processors/add_kubernetes_metadata/matchers.go#L42) might return empty string and it is not necessary identify any issue
so if there is no matchers configuration defined - we will get this debug message printed for each event, even though it is expected that index == "", correct?

Reply:
I don't think that should happen. We add the index inside the `addPod` which is called for each handler in each watcher. I don't see why this error would happen. If the event is not kubernetes related, we already returned in this function right in the beginning.

Reply:
ok, in case there are not much possibilities to encounter this scenario - lets keep it for troubleshooting as it would identify the misconfiguration (either used indexer or matcher are incorrect)

Reply:
Ok, I have added it again in the latest commit.